scripts/ci_workflow_binding.py

```
@@ -0,0 +1,273 @@
+            raise BindingError(f"ci.yml line {line_number} skips a mapping level")
+        stack[depth:] = []
+        if depth and values.get(tuple(stack)) is not None:
+            raise BindingError(f"ci.yml line {line_number} nests below a scalar")
+        key = match[2]
+        raw_value = match[3]
+        if raw_value == "":
+            value = None
+        elif raw_value.startswith(" ") and raw_value[1:] and not raw_value[1:].startswith(("|", ">", "&", "*", "!", "'", '"')):
+            value = raw_value[1:]
+        else:
+            raise BindingError(f"ci.yml line {line_number} uses unsupported YAML syntax")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)